pytensor/tensor/slinalg.py

```diff
@@ -1669,6 +1669,99 @@ def block_diag(*matrices: TensorVariable):
     return _block_diagonal_matrix(*matrices)
 
 
+def _to_banded_form(A, kl, ku):
+    """
+    Convert a full matrix A to LAPACK banded form for gbmv.
+
+    Parameters
+    ----------
+    A: np.ndarray
+        (m, n) banded matrix with nonzero values on the diagonals
+    kl: int
+        Number of nonzero lower diagonals of A
+    ku: int
+        Number of nonzero upper diagonals of A
+
+    Returns
+    -------
+    ab: np.ndarray
+        (kl + ku + 1, n) banded matrix suitable for LAPACK
+    """
+    A = np.asarray(A)
+    m, n = A.shape
+    ab = np.zeros((kl + ku + 1, n), dtype=A.dtype, order="C")
+
+    for i, k in enumerate(range(ku, -kl - 1, -1)):
+        padding = (k, 0) if k >= 0 else (0, -k)
+        diag = np.pad(np.diag(A, k=k), padding)
+        ab[i, :] = diag
+
+    return ab
+
+
+_dgbmv = scipy_linalg.get_blas_funcs("gbmv", dtype="float64")
+_sgbmv = scipy_linalg.get_blas_funcs("gbmv", dtype="float32")
```

Comment:
This will cause import time overhead to PyTensor.

I'm okay paying the extra 3us at runtime instead since virtually nobody will ever use this (or use it in a case where they need those extra us)

Reply:
I thought about this as well. It won't stay in the final verison.

Reply:
You can exploit `prepare_node` and add the function to `node.tag`, which the perform method can then retrieve from. That's two attribute accesses instead of a string check / blas caching...

Reply:
Or you can sidestep `perform` and use `make_thunk` instead